### PR TITLE
Remove NVVM 3.4 and CTK 11.0 / 11.1 support

### PR DESCRIPTION
**Pull request by gmarkall**
This removes support for CUDA toolkits 11.0 and 11.1. A large side-effect of this is that there are no longer any supported toolkits that use NVVM 3.4, so all NVVM 3.4-related code and tests can be dropped.

I decided against removing support for Kepler (CC 3.5 and 3.7) devices for the 0.58 release, because keeping it doesn't seem to be introducing any additional burden, and as long as the user has an 11.x toolkit installed, their toolkit will support it. It might be just fine to keep supporting these for more than one release into the future, so I've changed the deprecation notice to avoid specifying when CC 3.5 and 3.7 support will be removed.

The CI configuration is updated to remove the 11.0 and 11.1 configurations. I needed one more gpuCI configuration to be able to test all supported NumPy versions with CUDA, so I've (somewhat arbitrarily) added 11.3 into the matrix.

### CODE REVIEW

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
gpuci run tests

File: numba/cuda/cudadrv/nvvm.py [hunk 1]
```diff
@@ -363,6 +354,7 @@ def get_log(self):
     (11, 8): ((3, 5), (9, 0)),
     (12, 0): ((5, 0), (9, 0)),
     (12, 1): ((5, 0), (9, 0)),
+    (12, 2): ((5, 0), (9, 0)),
```
Comment (stuartarchibald) [inline numba/cuda/cudadrv/nvvm.py:357]: This is adding 12.2 whereas this patch concerns moving the baseline to 11.2?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, this looks good on inspection. A local search of the code base for patterns that might match `3.4` or are related to `11.0/11.1` yields a couple of instances in the docs that might warrant adjustment:
https://github.com/numba/numba/blob/766f6d9b250ebf452c75590f753de0c003563541/docs/source/cuda/cooperative_groups.rst?plain=1#L108
https://github.com/numba/numba/blob/766f6d9b250ebf452c75590f753de0c003563541/docs/source/cuda/minor_version_compatibility.rst?plain=1#L10-L13

**Comment by gmarkall** [comment]
> Thanks for the patch, this looks good on inspection. A local search of the code base for patterns that might match `3.4` or are related to `11.0/11.1` yields a couple of instances in the docs that might warrant adjustment:

Thanks - I think these should stay the same - the max blocks count (1152) may vary with CUDA driver and device, so it's not particularly invalid to state that example. The point was to show that a user running the example might get a different number with a different driver or device.

The second example is also relevant - CUDA driver 450 (CUDA 11.0) is supported because it supported later toolkits through MVC - it's just the toolkits we're dropping support for (as we require "a driver sufficient for the toolkit version").

File: numba/cuda/cudadrv/nvvm.py (hunk 1 above)
Comment (gmarkall) [reply numba/cuda/cudadrv/nvvm.py:357]: Yes, since this is a one-liner to add with no changed support, I added this in.

Comment (stuartarchibald) [reply numba/cuda/cudadrv/nvvm.py:357]: OK, thanks, let's leave it as-is given support isn't changing.

**Comment by stuartarchibald** [comment]
RE: https://github.com/numba/numba/pull/9040#issuecomment-1651841755

Thanks for considering these @gmarkall, am happy to leave them based on your assessment.

**Comment by sklam** [comment]
BFID: `numba_smoketest_cuda_yaml_210`

**Comment by stuartarchibald** [comment]
> BFID: `numba_smoketest_cuda_yaml_210`

This "passed", the CUDA 11.1 builds failed as would be expected as this PR removes support.

**Pull request merged**